gammapy/modeling/models/cube.py

```diff
@@ -196,15 +196,34 @@ def evaluate_geom(self, geom, gti=None):
         energy = geom.get_axis_by_name("energy_true").center[:, np.newaxis, np.newaxis]
         value = self.spectral_model(energy)
 
-        if self.spatial_model is not None:
+        if self.spatial_model:
             value = value * self.spatial_model.evaluate_geom(geom.to_image())
 
-        if self.temporal_model is not None:
+        if self.temporal_model:
             integral = self.temporal_model.integral(gti.time_start, gti.time_stop)
             value = value * np.sum(integral)
 
         return value
 
+    def integrate(self, geom, gti=None):
+        """Integrate model on `~gammapy.maps.Geom`."""
+        energy = geom.get_axis_by_name("energy_true").edges
+        value = self.spectral_model.integral(
+            energy[:-1], energy[1:], intervals=True
+        ).reshape((-1, 1, 1))
+
+        if self.spatial_model and not isinstance(geom, RegionGeom):
+            # TODO: integrate spatial model over region to correct for
+            #  containment
+            geom_image = geom.to_image()
+            value = value * self.spatial_model.integrate(geom_image).quantity
+
+        if self.temporal_model:
+            integral = self.temporal_model.integral(gti.time_start, gti.time_stop)
```

Comment:
Perhaps we should uniform naming between `.integrate` and `.integral` at some point?

Reply:
The API is different, because one takes `Geom` object and returns a `Map`, the other just works with numpy arrays. I could rename `.integrate()` to `.integrate_geom()` to avoid confusion?

Reply:
I've done the rename for now...